cpp/src/gandiva/function_registry_datetime.cc

```diff
@@ -83,7 +83,9 @@ std::vector<NativeFunction> GetDateTimeFunctionRegistry() {
       NativeFunction("extractDay", {}, DataTypeVector{day_time_interval()}, int64(),
                      kResultNullIfNull, "extractDay_daytimeinterval"),
 
-      DATE_TYPES(LAST_DAY_SAFE_NULL_IF_NULL, last_day, {})};
+      DATE_TYPES(LAST_DAY_SAFE_NULL_IF_NULL, last_day, {}),
+
+      DATE_TYPES(TO_TIMESTAMP_SAFE_NULL_IF_NULL, to_timestamp, {})};
```

Comment:
If the function converts number of seconds to timestamp, than the argument should be int32/int64/float/double. Date/timestamp as arguments would be incorrect as they are already in millis.